instrumentation/servlet/servlet-5.0/tomcat-testing/src/test/java/io/opentelemetry/javaagent/instrumentation/servlet/v5_0/tomcat/TomcatServlet5Test.java

```diff
@@ -0,0 +1,220 @@
+import io.opentelemetry.javaagent.instrumentation.servlet.v5_0.AbstractServlet5Test;
+import io.opentelemetry.sdk.testing.assertj.SpanDataAssert;
+import io.opentelemetry.sdk.testing.assertj.TraceAssert;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.testing.internal.armeria.common.AggregatedHttpRequest;
+import io.opentelemetry.testing.internal.armeria.common.AggregatedHttpResponse;
+import jakarta.servlet.Servlet;
+import java.io.File;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.catalina.Context;
+import org.apache.catalina.LifecycleException;
+import org.apache.catalina.core.StandardHost;
+import org.apache.catalina.startup.Tomcat;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+public abstract class TomcatServlet5Test extends AbstractServlet5Test<Tomcat, Context> {
+
+  private static final ServerEndpoint ACCESS_LOG_SUCCESS =
+      new ServerEndpoint(
+          "ACCESS_LOG_SUCCESS",
+          "success?access-log=true",
+          SUCCESS.getStatus(),
+          SUCCESS.getBody(),
+          false);
+  private static final ServerEndpoint ACCESS_LOG_ERROR =
+      new ServerEndpoint(
+          "ACCESS_LOG_ERROR",
+          "error-status?access-log=true",
+          ERROR.getStatus(),
+          ERROR.getBody(),
+          false);
+  private final TestAccessLogValve accessLogValue = new TestAccessLogValve();
+
+  @Override
+  protected void configure(HttpServerTestOptions options) {
+    super.configure(options);
+    options.setExpectedException(new IllegalStateException(EXCEPTION.getBody()));
+    options.setContextPath("/tomcat-context");
+    options.setTestError(testError());
+  }
+
+  public boolean testError() {
+    return false;
+  }
+
+  @Override
+  protected SpanDataAssert assertResponseSpan(
+      SpanDataAssert span, SpanData parentSpan, String method, ServerEndpoint endpoint) {
+    if (NOT_FOUND.equals(endpoint)) {
+      span.satisfies(s -> assertThat(s.getName()).matches(".*\\.sendError"))
+          .hasKind(SpanKind.INTERNAL)
+          .hasParent(parentSpan);
+    }
+    return super.assertResponseSpan(span, parentSpan, method, endpoint);
+  }
+
+  @Override
+  protected boolean hasResponseSpan(ServerEndpoint endpoint) {
+    return endpoint == NOT_FOUND || super.hasResponseSpan(endpoint);
+  }
+
+  @Override
+  protected Tomcat setupServer() throws Exception {
+    Tomcat tomcatServer = new Tomcat();
+
+    File baseDir = Files.createTempDirectory("tomcat").toFile();
+    baseDir.deleteOnExit();
```

Comment:
Not really caused by the conversion, but for this to work the directory would need to be empty.

Reply:
Hopefully fixed with https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12455